pool/pool.go

```diff
@@ -30,284 +31,587 @@ import (
+	x.timeout = timeout
+}
+
+// SetPollInterval specifies the interval, once it will check the completion of the operation.
+func (x *ContainerPollingParams) SetPollInterval(tick time.Duration) {
+	x.pollInterval = tick
 }
 
 // DefaultPollingParams creates ContainerPollingParams with default values.
 func DefaultPollingParams() *ContainerPollingParams {
 	return &ContainerPollingParams{
-		CreationTimeout: 120 * time.Second,
-		PollInterval:    5 * time.Second,
+		timeout:      120 * time.Second,
+		pollInterval: 5 * time.Second,
 	}
 }
 
-// AddNode adds address/weight pair to node PoolBuilder list.
-func (pb *Builder) AddNode(address string, priority int, weight float64) *Builder {
-	pb.nodeParams = append(pb.nodeParams, NodeParam{
-		address:  address,
-		priority: priority,
-		weight:   weight,
-	})
-	return pb
+type clientPack struct {
+	client  client
+	healthy bool
+	address string
 }
 
-// Build creates new Pool based on current PoolBuilder state and options.
-func (pb *Builder) Build(ctx context.Context, options *BuilderOptions) (*Pool, error) {
-	if len(pb.nodeParams) == 0 {
-		return nil, errors.New("no NeoFS peers configured")
-	}
+type prmCommon struct {
+	defaultSession bool
+	verb           sessionv2.ObjectSessionVerb
+	addr           *address.Address
 
-	nodesParams := make(map[int]*NodesParam)
-	for _, param := range pb.nodeParams {
-		nodes, ok := nodesParams[param.priority]
-		if !ok {
-			nodes = &NodesParam{priority: param.priority}
-		}
-		nodes.addresses = append(nodes.addresses, param.address)
-		nodes.weights = append(nodes.weights, param.weight)
-		nodesParams[param.priority] = nodes
-	}
+	key    *ecdsa.PrivateKey
+	btoken *token.BearerToken
+	stoken *session.Token
+}
 
-	for _, nodes := range nodesParams {
-		nodes.weights = adjustWeights(nodes.weights)
-		options.nodesParams = append(options.nodesParams, nodes)
-	}
+func (x *prmCommon) useDefaultSession() {
+	x.defaultSession = true
+}
 
-	sort.Slice(options.nodesParams, func(i, j int) bool {
-		return options.nodesParams[i].priority < options.nodesParams[j].priority
-	})
+func (x *prmCommon) useAddress(addr *address.Address) {
+	x.addr = addr
+}
 
-	return newPool(ctx, options)
+func (x *prmCommon) useVerb(verb sessionv2.ObjectSessionVerb) {
+	x.verb = verb
 }
 
-type clientPack struct {
-	client  Client
-	healthy bool
-	address string
+// UseKey specifies private key to sign the requests.
+// If key is not provided, then Pool default key is used.
+func (x *prmCommon) UseKey(key *ecdsa.PrivateKey) {
+	x.key = key
 }
 
-type CallOption func(config *callConfig)
+// UseBearer attaches bearer token to be used for the operation.
+func (x *prmCommon) UseBearer(token *token.BearerToken) {
+	x.btoken = token
+}
 
-type callConfig struct {
-	useDefaultSession bool
-	verb              sessionv2.ObjectSessionVerb
-	addr              *address.Address
+// UseSession specifies session within which object should be read.
```

Comment:
```suggestion
// UseSession specifies session within which object should be processed.
```
since `prmCommon` is embedded to non-read operations, too.

Reply:
Oh, it is embedded to non-object operations, too. So it could be:
```
// UseSession specifies session within which operation should be performed
```

Reply:
Oh, it is embedded to non-object operations, too. So it could be:
```
// UseSession specifies session within which operation should be performed
```